### PR TITLE
Explore: display more meaningful call to action when no datasource configured

### LINKED ISSUE #15533: CTA for adding a datasource when going to explore
2019-02-19-124725_524x130_scrot

> As a user I would like to be able to click a link/button to go directly to go to the "Add datasource" view

### PR DESCRIPTION
**Pull request by dprokop**
Fixes https://github.com/grafana/grafana/issues/15533

Displays call to action card that allows quick datasource configuration (me and @Ijin08 discussed the placement of the card and decided that left alignment, same as cheatsheet, is the besrt one - any opinion on that @davkal?): 

image

Also, fixed issue with DatasourceSrv not being updated after datasource was deleted.

### CODE REVIEW

File: public/app/features/explore/NoDataSourceCallToAction.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import React, { useContext } from 'react';
+import { css } from 'emotion';
+import { ThemeContext, ExtraLargeLinkButton, CallToActionCard } from '@grafana/ui';
+
+export const NoDataSourceCallToAction = () => {
+  const theme = useContext(ThemeContext);
+
+  const message = 'Please add a datasource that supports Explore (e.g., Prometheus)';
```
Comment (davkal) [inline public/app/features/explore/NoDataSourceCallToAction.tsx]: All datasources are supported now, so we can rephrase this:
"Explore requires at least one data source. Once you have added a data source, you can query it here."

File: public/app/features/explore/NoDataSourceCallToAction.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  const footer = (
+    <>
+      <i className="fa fa-rocket" />
+      <> ProTip: You can also define data sources through configuration files. </>
+      <a
+        href="http://docs.grafana.org/administration/provisioning/#datasources?utm_source=grafana_ds_list"
```
Comment (davkal) [inline public/app/features/explore/NoDataSourceCallToAction.tsx]: Nice protip! Also +1 on the utm markers.

**Review by davkal — APPROVED** [review]
Small text change needed, otherwise LGTM!

File: public/app/features/explore/NoDataSourceCallToAction.tsx (hunk 2 above)
Comment (dprokop) [reply public/app/features/explore/NoDataSourceCallToAction.tsx]: This is actually copied from data sources list. we need to update the utm_source here.

**Pull request merged**